modules/stochastic_tools/include/utils/BootstrapCalculators.h

```diff
@@ -74,30 +88,219 @@ class Percentile : public BootstrapCalculator<InType, OutType>
 {
 public:
   using BootstrapCalculator<InType, OutType>::BootstrapCalculator;
-  virtual std::vector<OutType> compute(const InType &, const bool) const override;
+  virtual std::vector<OutType> compute(const InType &, const bool) override;
 };
 
 /*
- * Implement BCa method of Efron and Tibshirani (2003), Chapter 14.
+ * This is placeholder class for BCa with general data types. Will throw an error if
+ * used. Actual implementation is with OutType = Real which can be seen below.
  */
 template <typename InType, typename OutType>
 class BiasCorrectedAccelerated : public BootstrapCalculator<InType, OutType>
 {
 public:
   using BootstrapCalculator<InType, OutType>::BootstrapCalculator;
-  virtual std::vector<OutType> compute(const InType &, const bool) const override;
+  virtual std::vector<OutType> compute(const InType &, const bool) override
+  {
+    mooseError(
+        "Cannot compute bias corrected accelerated statistics with calculator output value type ",
+        MooseUtils::prettyCppType<OutType>(),
+        ".");
+  }
+};
+
+/*
+ * Implement BCa method of Efron and Tibshirani (2003), Chapter 14.
+ */
+template <typename InType>
+class BiasCorrectedAccelerated<InType, Real> : public BootstrapCalculator<InType, Real>
+{
+public:
+  using BootstrapCalculator<InType, Real>::BootstrapCalculator;
+  virtual std::vector<Real> compute(const InType &, const bool) override;
 
 private:
   // Compute the acceleration, see Efron and Tibshirani (2003), Ch. 14, Eq. 14.15, p 186.
-  Real acceleration(const InType &, const bool) const;
+  Real acceleration(const InType &, const bool);
```

Comment:
Why is `const` being removed. Computing things for one InType should not change the state of the calculator? You should be able to call these as much as you want without any worry something is altering the calculation.

Reply:
I am seeing why this is done now, the Calculator::initialize/finalize. Is there some way we can assert the calculator state before resampleAct is called?

Reply:
I think I have an idea. I can make enum like this:
```cpp
enum StochasticTools:CalculatorState
{
  NONE;
  INITIALIZED;
  FINALIZED;
};
```
And have member variable in `Calculator`:
```cpp
  CalculatorState _state = CalculatorState::NONE;
```

And it will error if:
- `update` or `finalize` is called and `_state != CalculatorState::INITIALIZED`
- `get` is called and `_state != CalculatorState::FINALIZED`

In order to make the calculator `const` all the member variable would have to be `mutable`.  This might be okay with this state check.